src/server/server.ts

```diff
@@ -69,233 +74,121 @@ const ndlaMiddleware = [
-  const state = typeof req.query.state === "string" ? req.query.state : "/";
-  const redirect = constructNewPath(state, req.params.lang);
-  res.setHeader("Cache-Control", "private");
-
-  if (!feideToken?.["id_token"] || typeof state !== "string") {
-    return sendInternalServerError(res);
-  }
-  try {
-    const logoutUri = await feideLogout(req, redirect, feideToken["id_token"]);
-    return res.redirect(logoutUri);
-  } catch (_) {
-    return await sendInternalServerError(res);
-  }
-});
-
-app.get("/logout/session", (req: Request, res: Response) => {
-  res.clearCookie("feide_auth", { domain: `.${config.feideDomain}` });
-  const state = typeof req.query.state === "string" ? req.query.state : "/";
-  const { basepath, basename } = getLocaleInfoFromPath(state);
-  const wasPrivateRoute = privateRoutes.some((r) => matchPath(r, basepath));
-  const redirect = wasPrivateRoute ? constructNewPath("/", basename) : state;
-  res.setHeader("Cache-Control", "private");
-  return res.redirect(redirect);
-});
-
-app.get("/:lang?/subjects/:path(*)", ndlaMiddleware, (req: Request, res: Response, _next: NextFunction) => {
-  const { lang, path } = req.params;
-  res.redirect(301, lang ? `/${lang}/${path}` : `/${path}`);
-});
-
-export async function sendInternalServerError(res: Response) {
-  if (res.getHeader("Content-Type") === "application/json") {
-    res.status(INTERNAL_SERVER_ERROR).json("Internal server error");
-  } else {
-    const { data } = await errorRoute();
-    res.status(INTERNAL_SERVER_ERROR).send(data);
-  }
-}
-
-function sendResponse(res: Response, data: any, status = OK) {
-  if (status === MOVED_PERMANENTLY || status === TEMPORARY_REDIRECT) {
-    res.writeHead(status, data);
-    res.end();
-  } else if (status === GONE) {
-    res.status(status).send(data);
-  } else if (res.getHeader("Content-Type") === "application/json") {
-    res.status(status).json(data);
   } else {
-    res.status(status).send(data);
+    const { helmetContext, htmlContent, styles, data } = response.data;
+    const html = template
+      .replace('data-html-attributes=""', helmetContext?.helmet.htmlAttributes.toString() ?? "")
+      .replace("<!--TITLE-->", helmetContext?.helmet.title.toString() ?? "")
+      .replace("<!--META-->", helmetContext?.helmet.meta.toString() ?? "")
+      .replace("<!--LINK-->", helmetContext?.helmet.link.toString() ?? "")
+      .replaceAll("__ENVIRONMENT__", config.ndlaEnvironment === "dev" ? "test" : config.ndlaEnvironment)
+      .replace("<!--SCRIPT-->", helmetContext?.helmet.script.toString() ?? "")
+      .replace("<!--STYLES-->", styles ?? "")
+      .replace('data-body-attributes=""', helmetContext?.helmet.bodyAttributes.toString() ?? "")
+      .replace("REPLACE_ME", htmlContent)
+      .replace(
+        "<!--DATA-->",
+        serialize({
+          ...data,
+          config: {
+            ...data?.config,
+            isClient: true,
+          },
+        }),
+      );
```

Comment:
Klarer vi å dra templatene ut i funksjoner eller lignende for å slippe å iterere strengen så mange ganger pr request?